support/copyPluginDependencies.js

```diff
@@ -23,7 +27,25 @@ for (const [depType, deps] of Object.entries(depsConfig)) {
 		files.forEach((file) => {
 			const filePath = path.join('../node_modules', dep, file);
 
-			fileContent += fs.readFileSync(filePath);
+			if (depType === 'js') {
+				const code = fs.readFileSync(filePath);
+
+				const ast = parser.parse(code.toString());
+
+				traverse(ast, {
+					enter(path) {
+						if (path.node.name === 'define') {
```

Comment:
You could make this check a bit stricter... like:

```js
if (
    path.node.type === 'Identifier' &&
    path.node.name === 'define'
)
```

or better still:

```js
if (
    path.node.type === 'UnaryExpression' &&
    path.node.operator === 'typeof' &&
    path.node.argument.type === 'Identifier' &&
    path.node.argument.name === 'define'
)
```

The idea of using an AST-based transform is to make it precise, like a scalpel, so that you don't edit something that you don't mean to edit. So make it as precise as you can without it being brittle in the opposite sense.

To increase our confidence in this change, it would be good to grab a before and after diff just to "prove" it's not editing anything we don't want it to. Even as a once-off manual check pasted into the PR, that's still valuable

Reply:
Thanks for the suggestion @wincent, that's what I meant by 

> if someone can give me some pointers we could probably remove the entire if
but for the moment this fixes the issue.

I'm sure there's a way to make this more robust, but this is already a good start.

Here are some screenshots of the `ckeditor/plugins/codemirror/vendors/vendors.js` **before** (on the left) and **after** (on the right) 


![](https://user-images.githubusercontent.com/5572/100324273-4b1b2680-2fc7-11eb-8e80-9a734171145f.png)

![](https://user-images.githubusercontent.com/5572/100324287-51110780-2fc7-11eb-8493-158192b164cd.png)

![](https://user-images.githubusercontent.com/5572/100324304-579f7f00-2fc7-11eb-9ed3-ed56721d55c1.png)